### PR TITLE
Add syndicate sleeper agents random event

### PR DESCRIPTION
**Pull request by Dutch-VanDerLinde**
revival of #23896 

## About the PR
adds a sleeper agent event which selects 1-2 people to be traitors, raising the alert level to blue (if it's green), and sending an announcement saying that an enemy communication has been intercepted

## Why / Balance
adds more game variety

someone who has gained command's trust as a non-traitor then suddenly becomes a traitor can cause very interesting scenarios and conflict

## Technical details
adds ``AlertLevelInterceptionRule`` which is a gamerule that sets the station's alert level to the given ``Prototype`` if the alert level is green

## Media

https://github.com/space-wizards/space-station-14/assets/45323883/dcb6c44f-7213-4d6b-8ce2-5a34ea111d1e

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl: Vasilis, Dutch-VanDerLinde
- add: A new event has been added, syndicate sleeper agents, which raises the alert level and selects 1-2 players to be traitors rarely throughout the round.

### CODE REVIEW

**Comment by Plykiya** [comment]
does there really need to be a limit on the occurrences?

**Comment by mirrorcult** [comment]
agree that a limit on occurrences isnt necessary

**Comment by Dutch-VanDerLinde** [comment]
tweaks, instead of only being able to happen once it takes an hour to happen again
and it can occur as early as 25 minutes now

**Comment by FungiFellow** [comment]
War Ops but you just wait for sleeper agents before attacking :)

File: Content.Server/AlertLevel/AlertLevelSystem.cs [hunk 1]
```diff
@@ -96,6 +96,16 @@ private void OnPrototypeReload(PrototypesReloadedEventArgs args)
...
+    {
+        if (!Resolve(station, ref alert))
+        {
+            return string.Empty;
+        }
+
+        return alert.CurrentLevel;
+    }
+
     public float GetAlertLevelDelay(EntityUid station, AlertLevelComponent? alert = null)
```
Comment (metalgearsloth) [inline Content.Server/AlertLevel/AlertLevelSystem.cs:109]: Same here.

File: Content.Server/StationEvents/Components/AlertLevelInterceptionRuleComponent.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+[RegisterComponent, Access(typeof(AlertLevelInterceptionRule))]
+public sealed partial class AlertLevelInterceptionRuleComponent : Component
+{
+    /// <summary>
+    /// Alert level to set the station to when the event starts.
+    /// </summary>
+    [DataField]
+    public string AlertLevel = "blue";
+}
```
[the remark this answers is not in the document]
Comment (Dutch-VanDerLinde) [reply Content.Server/StationEvents/Components/AlertLevelInterceptionRuleComponent.cs:15]: no they aren't, it caused an error if i did AlertLevelprototype

[the remark this answers is not in the document]
Comment (Dutch-VanDerLinde) [reply Content.Server/StationEvents/Components/AlertLevelInterceptionRuleComponent.cs:15]: the alert levels are just a string dictionary in the AlertLevelPrototype
image

**Comment by Cojoke-dot** [comment]
So it is essentially a late-join syndicate that tells the whole station that there are syndicates on any round type? That feels like it could easily ruin the balance for non-syndicate rounds. ope, that person behind the captain just became a syndicate and is now rushing the Nukies in an elite suit and e-sword in trade for the captain's pistol. 

Gaining the trust of command and then betraying them after you become an antag 30 minutes in is... meh. You lose the fun of lying to gain their trust and tricking them into a private place. You lose a lot of time to plan too, resulting in a lot more boring shuttle bombings from people who were working on a project and were then randomly selected to be antag after the shift is almost finished.

**Comment by XxSWAG-MASTER-OVERLORDxX** [comment]
i like the idea, but i dislike the automatic code blue

**Comment by Hmeister-real** [comment]
I feel like it should be an automatic yellow instead, because sec are going to start immediately going ballistic if it isn't blue already and search everyone. but overall good PR

**Comment by ilyamikcoder** [comment]
how are enemy communications a structural threat though

it will be especially confusing, especially on MRP, where CE will be the main pal after the captain according to the alert procedure

**Comment by Dutch-VanDerLinde** [comment]
> I feel like it should be an automatic yellow instead, because sec are going to start immediately going ballistic if it isn't blue already and search everyone. but overall good PR

yellow does not make any sense

**Comment by Dutch-VanDerLinde** [comment]
> i like the idea, but i dislike the automatic code blue

knowing there's traitors aboard isn't really that dangerous of information, compared to nukies since

1. you don't know who the traitors are
2. traitors have varying objectives

to be honest, it'll most likely be code blue already by the time this event happens

**Comment by Simyon264** [comment]
It would be cool if it was only like a 70% chance to trigger agents and 30% just does nothing in order to confuse security.

**Comment by Dutch-VanDerLinde** [comment]
> It would be cool if it was only like a 70% chance to trigger agents and 30% just does nothing in order to confuse security.

yeah i tried to do this originally
either 0 traitors get selected, or 1-2 traitors get selected
but it didn't seem to work so i just made it 1-2

**Comment by FungiFellow** [comment]
I think the Automatic Code Blue is perfectly fine, Announcements do often get overlooked but having blue both gives security the ability to be more vigilant and saves command the time of walking over to the comms console.

**Comment by ilyamikcoder** [comment]
yeah i feel like if it won't set the alert automatically, command will just set it immediately, so why not

**Comment by FairlySadPanda** [comment]
This only will select from those with tot selected, right? The pool of who can be tot is too low as-is, especially on MRP.

**Comment by Dutch-VanDerLinde** [comment]
> This only will select from those with tot selected, right? The pool of who can be tot is too low as-is, especially on MRP.

Yes only with the traitor preference

**Comment by deltanedas** [comment]
if you dont want to be traitor it shouldnt pick you shrimple as

**Comment by Dutch-VanDerLinde** [comment]
Also wdym tot pool is low a ton of people have syndie enabled

**Comment by FairlySadPanda** [comment]
> if you dont want to be traitor it shouldnt pick you shrimple as

As said I'm worried the problem is the inverse: it means you're more likely to become a traitor if you have it selected.

I'm not personally a fan of volunteering for antag status as-is (if you're playing a social deduction game it sort of ruins the point if you metaknowledge that half the crew are literally never antag) but it makes no sense here where it's not a round progression antag role.

**Comment by FairlySadPanda** [comment]
Sidebar, another issue: this will cause more shuttlebombs and late-round shitter behaviour.

A roundstart traitor is going to have on average something like 50% more time to complete their objectives than a midround traitor.

**Comment by Dutch-VanDerLinde** [comment]
> Sidebar, another issue: this will cause more shuttlebombs and late-round shitter behaviour. Which is unwelcome on MRP from my research.
> 
> 
> 
> A roundstart traitor is going to have on average something like 2x more time to complete their objectives than a midround traitor.

Yes, this is for every midround antag

Ninjas will have less time to complete their objectives because they spawn midround

Ideally, when dynamic comes this event should only happen if the station is doing well (so the traitor(s) that get selected aren't picked when the station is exploding)

**Comment by FairlySadPanda** [comment]
Yes, midrounds always have less time. This means that design and balance of midround-only antags (ninja, loneop, pizza critter (TBC), ~exterminator~) factors that in.

Just dropping one to two full-strength tots in randomly I think causes balance problems.

Another problem: can an II, HR, regular rev, someone with zombie disease etc get tot as well?

A regular rev suddenly getting 20TC would be a huge buff to Revs. for example.

**Comment by Dutch-VanDerLinde** [comment]
> Yes, midrounds always have less time. This means that design and balance of midround-only antags (ninja, loneop, pizza critter (TBC), ~exterminator~) factors that in.
> 
> 
> 
> Just dropping one to two full-strength tots in randomly I think causes balance problems.
> 
> 
> 
> Another problem: can an II, HR, regular rev, someone with zombie disease etc get tot as well? Can a Closet Skeleton? Rat King?
> 
> 
> 
> A regular rev suddenly getting 20TC would be a huge buff to Revs. for example.

People who are already antags shouldn't be selected for sleeper agents if selection code is working correctly

**Comment by Dutch-VanDerLinde** [comment]
Perhaps the traitors should be nerfed slightly, they get less TC but easier and less destructive objectives

**Comment by K-Dynamic** [comment]
> Perhaps the traitors should be nerfed slightly, they get less TC but easier and less destructive objectives

Should we differentiate them in the end screen? Like 'sleeper agent' instead of 'traitor'?

**Comment by muburu** [comment]
it'd be neat if the ninja could call this in as a threat via the comms console, only like 2 other mid-rounds can be called in from ninja so we'd get some nice variety from that

**Comment by Dutch-VanDerLinde** [comment]
> > Perhaps the traitors should be nerfed slightly, they get less TC but easier and less destructive objectives
> 
> 
> 
> Should we differentiate them in the end screen? Like 'sleeper agent' instead of 'traitor'?

They're still syndies so not really any purpose to that

**Comment by Dutch-VanDerLinde** [comment]
> it'd be neat if the ninja could call this in as a threat via the comms console, only like 2 other mid-rounds can be called in from ninja so we'd get some nice variety from that

Not sure, rev and dragon are minor antagonists compared to syndies

**Comment by deltanedas** [comment]
rev without abusing bugs has next to no killing power, more of an annoyance than an antag

dragon can be round ending if robust, syndies can fumble or troll everyone

**Comment by Dutch-VanDerLinde** [comment]
> rev without abusing bugs has next to no killing power, more of an annoyance than an antag
> 
> 
> 
> dragon can be round ending if robust, syndies can fumble or troll everyone

Dragons kinda weak if it doesn't hide in maints for half the time and make carp army

**Comment by FungiFellow** [comment]
Dragons are entirely dependent on spawn location, God forbid people see your spawn and you get rushed by 5 tiders and a seccie before you have a single carp out

**Comment by Hmeister-real** [comment]
> > if you dont want to be traitor it shouldnt pick you shrimple as
> 
> As said I'm worried the problem is the inverse: it means you're more likely to become a traitor if you have it selected.
> 
> I'm not personally a fan of volunteering for antag status as-is (if you're playing a social deduction game it sort of ruins the point if you metaknowledge that half the crew are literally never antag) but it makes no sense here where it's not a round progression antag role.

If your meta-remembering peoples character after they say they don't have tot enabled, that's an admin issue, i play the game like everyone i meet is a new person every round

**Comment by Tynobeard** [comment]
> If your meta-remembering peoples character after they say they don't have tot enabled, that's an admin issue, i play the game like everyone i meet is a new person every round

Pretending and RPing like everyone is elegible for traitor is makes way less sense then just having everyone be elegible for traitor. How can an admin find out what someone is thinking about someone else and punish them accordingly? Having the meta knoledge that half the station was uneffected by sleeper agent is using it weather consciously or not imo.

**Pull request merged**